cmd/ctl/cmd/cmd.go

```diff
@@ -39,6 +40,12 @@ import (
 )
 
 func NewCertManagerCtlCommand(in io.Reader, out, err io.Writer, stopCh <-chan struct{}) *cobra.Command {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
We should update the SignalHandler in https://github.com/jetstack/cert-manager/blob/26afeceb2b8f7499269c16670bfdd0f884275cd6/pkg/util/cmd/signal.go#L29

to instead return a `context.Context`. This context can then be created in `main.go` for ctl. For the other commands which accept a `<- chan struct{}`, we can instead pass in `ctx.Done()`.

Reply:
Alternative, we have `ContextWithStopCh` here: https://github.com/jetstack/cert-manager/blob/26afeceb2b8f7499269c16670bfdd0f884275cd6/pkg/util/context.go#L23-L26 😄

Reply:
Thank you for the review!

I have changed it to use `ContextWithStopCh` instead.

Would it make sense to change also [this part](https://github.com/jetstack/cert-manager/blob/master/cmd/cainjector/main.go#L40) in the codebase to use `ContextWithStopCh`?

Reply:
Yep, makes sense to me!

Reply:
> Would it make sense to change also this part in the codebase to use ContextWithStopCh?

I added that change to this PR in a separate commit. I hope that is okay, because it is really small- otherwise I could create a separate issue.